src/main/scala/config/KafkaSettings.scala

```diff
@@ -1,21 +1,28 @@
 package config
 
-import scala.collection.Map
 import scala.collection.JavaConverters._
 import com.typesafe.config.Config
 import org.apache.kafka.clients.CommonClientConfigs
 import org.apache.kafka.clients.admin.AdminClient
 
-case class KafkaSettings(address: String, commandConfig: String) {
```

Comment:
Would something like this make it more readable? Up to you.
```
case class KafkaSettings(address: String, commandConfig: String, adminClientConfig: Map[String, AnyRef]) {
  lazy val adminClient: AdminClient = AdminClient.create(adminClientConfig.asJava)
}

object KafkaSettings {
  def apply(config: Config): KafkaSettings = {
    val adminClientConfig = config.getObject("kafka.admin-client")
      .withValue(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG, config.getValue("kafka.endpoint"))
      .unwrapped().asScala
      .filterKeys(org.apache.kafka.clients.admin.AdminClientConfig.configNames.contains)
      .toMap

    KafkaSettings(
      config.getString("kafka.endpoint"),
      config.getString("kafka.command.config"),
      adminClientConfig
    )
  }
```

Reply:
I had used `.unwrapped()` first, but that was causing the `security.protocol` key to be split (i.e. it was `security -> (protocol -> SSL)` as opposed to `security.protocol -> SSL`) and couldn't be read properly by Kafka when initiating the AdminClient

Reply:
@soceanainn the key needs to be quoted, I believe:

```
kafka {
  admin-client {
    "security.protocol" = "SSL"
  }
}
```

Reply:
@ioreskovic I wanted to avoid the need for quoting the key in these situations, so while the code below is a little bit clunkier, it avoids any unexpected issues when users are configuring the `AdminClient`

Reply:
I understand, but IIRC, this is a convention even in alpakka-kafka library where the end user is given the responsibility to provide correct keys.